pkg/pool-stable/contracts/ComposableStablePoolProtocolFees.sol

```diff
@@ -226,44 +231,59 @@ abstract contract ComposableStablePoolProtocolFees is
         // Note that the pre-join/exit invariant is *not* the invariant from the last join,
         // but computed from the balances before this particular join/exit.
 
-        uint256 protocolSwapFeePercentage = getProtocolFeePercentageCache(ProtocolFeeType.SWAP);
-
-        if (protocolSwapFeePercentage > 0) {
-            uint256 invariantGrowthRatio = (
-                postJoinExitInvariant > preJoinExitInvariant
-                    ? postJoinExitInvariant.sub(preJoinExitInvariant)
-                    : preJoinExitInvariant.sub(postJoinExitInvariant)
-            )
-                .divDown(preJoinExitInvariant);
-
-            // Compute the bpt ratio
-            uint256 bptGrowthRatio = (
-                postJoinExitSupply > preJoinExitInvariant
-                    ? postJoinExitSupply.sub(preJoinExitSupply)
-                    : preJoinExitSupply.sub(postJoinExitSupply)
-            )
-                .divDown(preJoinExitSupply);
-
-            // The difference between the invariant growth and bpt increase rates must be due to the
-            // balance change from this join/exit.
-            // Protocol fees due = (invariant growth / bpt increase - 1) * virtual supply * protocol fee %
-            // For instance, if the invariant growth is 1.05, and the bpt increase is 1.0475, with 1000 supply,
-            // and a protocol fee of 50%, we would mint (1.05/1.0475 - 1) * 1000 * 0.5 = 1.193 BPT.
-
-            if (invariantGrowthRatio > bptGrowthRatio) {
-                uint256 protocolFeeAmount = invariantGrowthRatio
-                    .divDown(bptGrowthRatio)
-                    .sub(FixedPoint.ONE)
-                    .mulDown(preJoinExitSupply)
-                    .mulDown(protocolSwapFeePercentage);
-
-                _payProtocolFees(protocolFeeAmount);
-            }
+        // `_payProtocolFeesBeforeJoinExit` paid protocol fees accumulated between the previous and current
+        // join or exit. This code pays any protocol fees due on a `current` joinSwap or exitSwap.
+        // The amp and rates are constant during a single transaction, so it doesn't matter if there
+        // is an ongoing amp change, and we can ignore yield.
+
+        // This represents the total value exchanged in this particular join-exit, including swap fees.
```

Comment:
```suggestion
        // This represents the total value exchanged in this particular join-exit, including swap fees.
        // Note that as swap fees are symmetric, we can swap the pre- and post-exit invariants and treat it like a join.
```

I think the naming of variables here is a bit of a hurdle to it being obvious that this is correct for exits. We could rename them to `*Change` but I think it might be best to just add a comment like the above maybe?

Reply:
Yeah; I went back and forth on that. It was "delta" for a while, but it's not really a delta in the sense that's used elsewhere (after dividing, anyway). I added a detailed comment, perhaps a bit long-winded.

Reply:
Ah yeah, I forgot about the division so my comment above isn't correct.